src/H5Dio.c

```diff
@@ -302,22 +315,25 @@ H5D__read(size_t count, H5D_dset_io_info_t *dset_info)
     if (io_skipped == count)
         HGOTO_DONE(SUCCEED);
 
-    /* Perform second phase of type info initialization */
-    if (H5D__typeinfo_init_phase2(&io_info) < 0)
-        HGOTO_ERROR(H5E_DATASET, H5E_CANTINIT, FAIL, "unable to set up type info (second phase)");
+    /* Initialize type info differently depending on if weŕe using the shared chunk cache or not */
```

Comment:
The shared chunk cache will need to perform its own type conversion initialization.